app/controllers/tutorials_controller.rb

```diff
@@ -1,6 +1,7 @@
 class TutorialsController < ApplicationController
   def show
     tutorial = Tutorial.find(params[:id])
+    four_oh_four unless tutorial.classroom == false || current_user
```

Comment:
Interesting way to render a 404!

Reply:
Yeah, the four_oh_four method came with the project, but I'm not sure it's the best way to render a 404 (it just looks like the app broke when it's invoked instead of showing a nicely-formatted 404). This is something we can discuss during eval tomorrow when we bring up technical debt that came with the app.